Replicator Panics If A System Job Is Registered With A Replicator Scaling Policy

**Description**

The system scheduler is used to register jobs that should be run on all clients that meet the job's constraints. The system scheduler is also invoked when clients join the cluster or transition into the ready state. This means that all registered system jobs will be re-evaluated and their tasks will be placed on the newly available nodes if the constraints are met.

If a job is registered with the `system` type and includes a Replicator scaling policy, a panic is triggered. 

There is no reason anyone should include a scaling policy on a system job, however, Replicator should not panic if someone does. 

**Steps to reproduce the issue:**
1. Register a Nomad job with type set to `system` (example job provided below)
2. Run Replicator agent with job scaling enabled `replicator agent`
3. Monitor Replicator and observe segmentation violation when Replicator checks to see if the job is in deployment. 

**Describe the results you expected:**
Replicator should not register a job as scalable if the job type is system. 

**Describe the results you received:**
Replicator registers the job as scalable and attempts to check if the job is currently in deployment. A segmentation violation causes a panic. 

**Additional information you deem important (e.g. issue happens only occasionally):**

**Replicator version:**
```
Replicator v1.1.0-dev
```
**Output of `consul version`:**

```
Consul v0.9.3
Protocol 2 spoken by default, understands 2 to 3 (agent will automatically use protocol >2 when speaking to compatible agents)
```

**Output of `nomad version`:**

```
Nomad v0.7.0-beta1
```

**Additional environment details:**

Example job file to reproduce error: 
```
job "example" {
  datacenters = ["dc1"]
  type = "system"

  update {
    max_parallel = 1
    min_healthy_time = "10s"
    healthy_deadline = "3m"
    auto_revert = false
    canary = 1
  }

  group "web" {
    count = 1

    meta {
      "replicator_cooldown"         = 15
      "replicator_enabled"          = true
      "replicator_max"              = 3
      "replicator_min"              = 1
      "replicator_notification_uid" = "REP1"
      "replicator_scalein_mem"      = 99
      "replicator_scalein_cpu"      = 99
      "replicator_scaleout_mem"     = 1
      "replicator_scaleout_cpu"     = 1
    }

    task "nginx" {
      driver = "docker"

      config {
        image = "nginxdemos/hello:latest"
      }

      resources {
        cpu    = 500 # 500 MHz
        memory = 256 # 256MB
        network {
          mbits = 10
        }
      }
    }
  }
}
```
**Debug log outputs from replicator:**

```
[INFO] command/agent: running version 1.1.0-dev
[INFO] command/agent: starting replicator agent...
[INFO] command/agent: replicator is running with cluster scaling globally disabled
[INFO] command/agent: replicator is running with job scaling globally enabled
[INFO] client/job_scaling_policies: added new policy for job example and group web
[INFO] client/consul: leadership lock successfully obtained at replicator/config/leader
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x169664b]

goroutine 13 [running]:
github.com/elsevier-core-engineering/replicator/client.(*nomadClient).IsJobInDeployment(0xc42020c008, 0xc420470308, 0x7, 0x1056700)
	/Users/westfaex/code/go/src/github.com/elsevier-core-engineering/replicator/client/job_scaling.go:183 +0xdb
github.com/elsevier-core-engineering/replicator/replicator.(*Runner).asyncJobScaling(0xc420461aa0, 0xc420284660)
	/Users/westfaex/code/go/src/github.com/elsevier-core-engineering/replicator/replicator/job_scaling.go:28 +0xd6
github.com/elsevier-core-engineering/replicator/replicator.(*Runner).jobScalingTicker(0xc420461aa0, 0xc420284660)
	/Users/westfaex/code/go/src/github.com/elsevier-core-engineering/replicator/replicator/runner.go:99 +0x16f
created by github.com/elsevier-core-engineering/replicator/replicator.(*Runner).Start
	/Users/westfaex/code/go/src/github.com/elsevier-core-engineering/replicator/replicator/runner.go:60 +0x220
```

--- a/client/job_scaling_policies.go
+++ b/client/job_scaling_policies.go
@@ -3,11 +3,13 @@ package client
 import (
 	"time"
 
+	nomad "github.com/hashicorp/nomad/api"
+	nomadStructs "github.com/hashicorp/nomad/nomad/structs"
+	"github.com/mitchellh/mapstructure"
+
 	"github.com/elsevier-core-engineering/replicator/helper"
 	"github.com/elsevier-core-engineering/replicator/logging"
 	"github.com/elsevier-core-engineering/replicator/replicator/structs"
-	nomad "github.com/hashicorp/nomad/api"
-	"github.com/mitchellh/mapstructure"
 )
 
 // JobWatcher is the main entry point into Replicators process of reading and
@@ -67,6 +69,12 @@ func (c *nomadClient) jobScalingPolicyProcessor(jobID string, scaling *structs.J
 	jobInfo, _, err := c.nomad.Jobs().Info(jobID, &nomad.QueryOptions{})
 	if err != nil {
 		logging.Error("client/job_scaling_policies: unable to call Nomad job info: %v", err)
+	}
+
+	// If the job type is incompatible with scaling, there is no need to
+	// process the configuration.
+	if *jobInfo.Type == nomadStructs.JobTypeSystem {
+		return
 	}
 
 	// It seems when a job is stopped Nomad notifies twice; once indicates the job

